File: 3704-count-partitions-with-even-sum-difference/3704-count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        s = sum(nums)
        left, cnt = 0, 0
        for i in range(len(nums) - 1):
            left += nums[i]
            if ((abs(s - 2 * left)) & 1) == 0:
                cnt += 1
        return cnt

File: 3704-count-partitions-with-even-sum-difference/3704-count-partitions-with-even-sum-difference.go
func countPartitions(nums []int) int {
    n, left, cnt := len(nums), 0, 0
    sum := 0
    for _, num := range nums {
        sum += num
    }
    for i := 0; i < n - 1; i++ {
        left += nums[i]
        temp := sum - 2 * left
        if temp & 1 == 0 {
            cnt++
        }
    }
    return cnt
}